src/js/popup/popup.js

```diff
@@ -531,6 +530,113 @@ async function enableSettingsPanel() {
+  [reportIssueToggle, reportIssueSettingsReturn, submissionSuccessContinue].forEach(e => {
+    e.addEventListener("click", () => {
+      document.body.classList.toggle("show-report-issue");
+      const eventLabel = document.body.classList.contains("show-report-issue") ? "opened-report-issue" : "closed-report-issue";
+      if (document.body.classList.contains("show-report-issue")) {
+        sendRelayEvent("Panel", "click", eventLabel);
+      }
+    });
+  });
+  const reportForm = document.querySelector(".report-issue-content");
+  setURLwithIssue();
+  showReportInputOtherTextField();
+  showSuccessReportSubmission();
+  reportForm.addEventListener('submit', handleReportIssueFormSubmission);
+}
+
+async function handleReportIssueFormSubmission(event) {
+  event.preventDefault();
+  const data = new FormData(event.target);
+  const reportData = Object.fromEntries(data.entries());
+
+  Object.keys(reportData).forEach(function(value) {
+    // Switch "on" to true
+    if (reportData[value] === "on") {
+      reportData[value] = true;
+    }
+    // Remove from report if empty string
+    if (reportData[value] === "") {
+      delete reportData[value];
+    }
+  });
+  // Clean URL data to add "http://" before it if the custom input doesn't contain a HTTP protocol
+  if (!(reportData.issue_on_domain.startsWith("http://") || reportData.issue_on_domain.startsWith("https://"))) {
+    reportData.issue_on_domain = "http://" + reportData.issue_on_domain;
+  }
+  await browser.runtime.sendMessage({
+    method: "postReportWebcompatIssue",
+    description: reportData
+  });
+}
+
+function showSuccessReportSubmission() {
+  const reportIssueSubmitBtn = document.querySelector(".report-issue-submit-btn");
+  const reportSuccess = document.querySelector(".report-success");
+  const reportContent = document.querySelector(".report-issue-content");
+  reportIssueSubmitBtn.addEventListener("click", () => {
+    reportSuccess.classList.remove("is-hidden");
+    reportContent.classList.add("is-hidden");
+  });
+}
+
+function isValidURL(string) {
+  // Disabled es-lint in the following line to accept regex
+  /* eslint-disable-next-line */
+  var expression = /(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/g;
+  const res = string.match(expression);
+  return (res !== null)
+}
+
+async function setURLwithIssue() {
+  // Add Site URL placeholder
+  const currentPage = (await getCurrentPage()).url;
+  const reportIssueSubmitBtn = document.querySelector(".report-issue-submit-btn");
+  const inputFieldUrl = document.querySelector('input[name="issue_on_domain"]');
+  reportIssueSubmitBtn.disabled = true;
+
+  // Allow for custom URL inputs
+  inputFieldUrl.addEventListener('input', () => {
+    reportIssueSubmitBtn.disabled = true;
+    // Ensure that the custom input is valid
+    if (isValidURL(inputFieldUrl.value)) {
+      reportIssueSubmitBtn.disabled = false;
+    }
+  });
```

Comment:
This is what checks for the validity of custom inputted URLs.

Reply:
Instead of adding custom styling, you could also outsource form validation to the browser by calling [`.setCustomValidity`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/setCustomValidity) (followed by `.reportValidity()`) on the URL input field, which should also prevent the form from being submitted. (You could also set `type="url" required` on the input field, I think, but that does require people to type `https://` themselves.)